Style using the Foundation framework

katherine-boost opened the pull request:
This makes the amount of custom CSS much smaller.

Also:
• Pulled out list item (i.e., actual grocery item which is part of a shopping list) into a partial. This was just for readability.

Addresses feedback in #3:
* Generally just use Foundation
* Use `rem` rather than `em` or `px`
    * Exceptions: `border-radius` (I think `px` makes sense for that), and the Foundation breakpoints which are in `em`.

I did have some problems with Foundation - the `XY Grid` was not working whatsoever; neither could I use the new breakpoint syntax (e.g. `@include breakpoint(medium only)`).  
Per its repo, the Rails gem appears to be up to date, so it's pretty puzzling.

### Mobile views

<img width="387" alt="shopping-list_homepage_mob" src="https://user-images.githubusercontent.com/43123718/45463795-c931e080-b761-11e8-99fe-63a4c5c9da31.png">
<img width="383" alt="shopping-list_list_mob" src="https://user-images.githubusercontent.com/43123718/45463796-c9ca7700-b761-11e8-8793-d7acb1d94bdd.png">

### Desktop views

<img width="1432" alt="shopping-list_homepage_dt" src="https://user-images.githubusercontent.com/43123718/45463896-1ada6b00-b762-11e8-9cac-c02488c561b9.png">
<img width="1434" alt="shopping-list_list_dt" src="https://user-images.githubusercontent.com/43123718/45463895-1ada6b00-b762-11e8-88f2-83fe5a303c75.png">

Okay it's still not very beautiful, but at least it's in Foundation now!

katherine-boost committed:
Style using the Foundation framework

This makes the amount of custom CSS much smaller.

Also:
• Pulled out list item (i.e., actual grocery item which is part of a shopping list) into a partial. This was just for readability.

tim-boost commented on app/assets/stylesheets/application.css.scss:
```diff
@@ -16,76 +16,77 @@
+
+/* CUSTOM VARIABLES */
+
 $boost-red: #B72039;
```
Could be worth making a _variables.scss or _colours.scss file to store colours and then import in that file above

katherine-boost replied:
Done in 82a524a

tim-boost commented on app/assets/stylesheets/application.css.scss:
```diff
@@ -16,76 +16,77 @@
 
+/* FOUNDATION BREAKPOINTS */
+
+$medium-start: 40.063em;
```
Could replace these with the ones from the foundation overrides file at some point

katherine-boost replied:
Done in next commit :)

tim-boost commented on app/assets/stylesheets/application.css.scss:
```diff
@@ -16,76 +16,77 @@
+
 body {
+  background-color: $boost-red;
   color: white;
```
Could make this a variable

tim-boost replied:
Or use $primary-color

katherine-boost replied:
do you mean make `background-color: $boost-red;` a variable in itself?

tim-boost commented on app/views/lists/show.html.erb:
```diff
@@ -44,16 +27,23 @@
+      <%= form.select(:category_id, options_for_select(@categories.map { |c| [c.name, c.id] })) %>
+    </div>
+    <div class="columns small-6 large-3">
+      <%= form.submit "Add item", class: "button small" %>
```
Better rails style to use single quotes here

katherine-boost replied:
Done in 82a524a (+ other instances)

tim-boost commented on app/views/lists/index.html.erb:
```diff
@@ -16,7 +20,7 @@
 </div>
 
-<%= link_to lists_path, method: :post, class: "nav-link" do %>
+<%= link_to lists_path, method: :post, class: "button" do %>
```
Could use single quotes here

katherine-boost replied:
Done in 82a524a

katherine-boost committed:
Styling: use Foundation variables for breakpoints

katherine-boost committed:
Separate SCSS file for colours; Rails single-quoting

Where HTML class names are passed as a parameter using class:, I've single-quoted them.
When written directly as HTML, I've kept the double quotes.

e.g.:
<%= link_to lists_path, method: :post, class: 'button' do %>
<i class="icon-right fi-plus"></i>
